### PR TITLE
add `isomorphism`

### PR DESCRIPTION
**Pull request by ThomasBreuer**
This is a follow-up to #1182.

- deprecate
  `isomorphic_fp_group(G::GAPGroup)`,
  `isomorphic_pc_group(G::GAPGroup)`,
  `isomorphic_perm_group(G::GAPGroup)`
  in favour of
  `isomorphism(::Type{FPGroup}, G::GAPGroup)`
  `isomorphism(::Type{PcGroup}, G::GAPGroup)`
  `isomorphism(::Type{PermGroup}, G::GAPGroup)`

- deprecate also (the undocumented) `isomorphic_group(::Type{T}, G::GAPGroup)`
  (does not occur in tests) in favour of `isomorphism`

- added
  `isomorphism(::Type{FPGroup}, A::GrpAbFinGen)`
  `isomorphism(::Type{PcGroup}, A::GrpAbFinGen)`
  `isomorphism(::Type{PermGroup}, A::GrpAbFinGen)`

- moved `_get_iso_function` and `isomorphic_group`
  from `sub.jl` to `homomorphisms.jl`

- added tests for new conversions

- adjusted tests to use `isomorphism` instead of the deprecated versions

- added (::Type{T})(G::GAPGroup)

### CODE REVIEW

**Comment by fieker** [comment]
the old versin is used - at least the Galois stuff has to be adjusted as well

File: src/Groups/homomorphisms.jl [hunk 1]
```diff
@@ -371,6 +371,33 @@ function isisomorphic(G::GAPGroup, H::GAPGroup)
...
+_get_iso_function(::Type{PcGroup}) = GAP.Globals.IsomorphismPcGroup
+
+
+"""
+    isomorphism(::Type{T}, G::GAPGroup) where T <: Union{FPGroup, PcGroup, PermGroup}
+
+Return an isomorphism from `G` to a group of type `T`.
+An exception is thrown if no such isomorphism exists.
+"""
+function isomorphism(::Type{T}, G::GAPGroup) where T <: Union{FPGroup, PcGroup, PermGroup}
```
Comment (fingolfin) [inline src/Groups/homomorphisms.jl:392]: Should we perhaps cache these isomorphisms, i.e.
```suggestion
@attr function isomorphism(::Type{T}, G::GAPGroup) where T <: Union{FPGroup, PcGroup, PermGroup}
```

**Review by fingolfin — APPROVED** [review]
Looks good to me. @fieker I've adjusted the Galois groups code, too

File: src/Groups/homomorphisms.jl (hunk 1 above)
Comment (ThomasBreuer) [reply src/Groups/homomorphisms.jl:392]: Caching is a good idea, but `isomorphism` takes two arguments, and `@attr` can deal only with unary functions.
(From this point of view, the old `isomorphic_fp_group` etc. were more suitable.)

File: docs/src/Groups/grouphom.md [hunk 2]
```diff
@@ -167,8 +167,7 @@ preimage(f::GAPGroupHomomorphism{S, T}, H::T) where S <: GAPGroup where T <: GAP
 ## Groups created by isomorphisms
```
Comment (fingolfin) [inline docs/src/Groups/grouphom.md:167]: I really don't like this section title. Though I also don't have a great alternative suggestion... perhaps just
```suggestion
## Group isomorphisms
```

File: src/Groups/homomorphisms.jl (hunk 1 above)
Comment (fingolfin) [reply src/Groups/homomorphisms.jl:392]: Yeah, we can't just use `@attr`, but we can still "manually" implement that caching. Anyway, not in this PR.

**Pull request merged**